onnxruntime/core/providers/cpu/controlflow/if.cc

```diff
@@ -417,6 +417,9 @@ Status IfImpl::Execute(const FeedsFetchesManager& ffm) {
 
   ORT_RETURN_IF_ERROR(status);
 
+  // block host until the stream is complete as fetches will be accessed from CPU
```

Comment:
please document that previously Fence is used here to guard the data access. now since Fence is retired, we need explicit sync. 
Actually i think it might be better to have a "sync stream" flag in "ExecuteSubGraph", and the caller decide whether to sync or not based on the needs.